### DISCUSSION TITLE
mxnet cpp-package example mlp_cpu.cpp does not compile MSVC 2019 and windows 10

### DISCUSSION THREAD

**Discussion by jbensabat** [apache/incubator-mxnet#19874] [Q&A]
Hello 
I managed to build mxnet libs and also the op.h file using the python script 
trying to compile the file mlp_cpu.cpp I get errors of undefined functions 

C:\software\mxnet\cpp-package\include\mxnet-cpp\executor.h(62,5): error C3861: 'MXExecutorForward': identifier not found
1>C:\software\mxnet\cpp-package\include\mxnet-cpp\executor.h(65,5): error C3861: 'MXExecutorOutputs': identifier not found
1>C:\software\mxnet\cpp-package\include\mxnet-cpp\executor.h(87,7): error C3861: 'MXExecutorBackward': identifier not found
1>C:\software\mxnet\cpp-package\include\mxnet-cpp\executor.h(89,7): error C3861: 'MXExecutorBackward': identifier not found
1>C:\software\mxnet\cpp-package\include\mxnet-cpp\executor.h(103,17): error C3861: 'MXExecutorFree': identifier not found
1>C:\software\mxnet\cpp-package\include\mxnet-cpp\executor.hpp(78,1): error C3861: 'MXExecutorBindEX': identifier not found
1>C:\software\mxnet\cpp-package\include\mxnet-cpp\executor.hpp(89,3): error C3861: 'MXExecutorOutputs': identifier not found
1>C:\software\mxnet\cpp-package\include\mxnet-cpp\executor.hpp(97,3): error C3861: 'MXExecutorPrint': identifier not found

I searched for these funntions and they do not seem to exist in version 1.7.0 
any help would be welcome
best
jac

## Description
(A clear and concise description of what the bug is.)

### Error Message
(Paste the complete error message. Please also include stack trace by setting environment variable `DMLC_LOG_STACK_TRACE_DEPTH=100` before running your script.)

## To Reproduce
(If you developed your own code, please provide a short script that reproduces the error. For existing examples, please provide link.)

### Steps to reproduce
(Paste the commands you ran that produced the error.)

1.
2.

## What have you tried to solve it?

1.
2.

## Environment

***We recommend using our script for collecting the diagnostic information with the following command***
`curl --retry 10 -s https://raw.githubusercontent.com/apache/incubator-mxnet/master/tools/diagnose.py | python3`

<details>
<summary>Environment Information</summary>

```
# Paste the diagnose.py command output here
```

</details>

**Comment by szha** [comment]
@jbensabat I think you might have mixed the use of v1.x and master branch. Could you make sure that you are building the library from the v1.x branch?

**Comment by jbensabat** [comment]
Hi 
thanks for your answer 
I downloaded it using git
"git clone --recursive https://github.com/apache/incubator-mxnet mxnet"
I am not sure it was version 1.7.0 
I was suggested to use this mode of downloading instead of downloading the archive (it did not build with CMAKE)
if this is the case how can I download version 1.7.0 in such a way that I can build ?
best
jac

**Comment by szha** [comment]
@jbensabat before building, you can check out the branch v1.7.x. Which instruction did you follow for building mxnet for cpp?

**Comment by jbensabat** [comment]
Hi
I tried to download the zip file for version 1.7.x but was unable to build using cmake.
then I was suggested to download using git 
"git clone --recursive https://github.com/apache/incubator-mxnet mxnet"
but then I do not know which version is installed.
the library builds but not the cpp-package 
I managed to create the op.h file but the functions mentioned above are missing 
best
jac

**Comment by szha** [comment]
Those functions are removed from the master branch which is for development of the 2.0 version. If there's any issue in using v1.7.x version for building, I think you can check out v1.x version and build from there.

**Comment by jbensabat** [comment]
I got that 
the question is how do  clone using git command above a specific version ?
best 
jac

**Comment by szha** [comment]
`git clone --recursive https://github.com/apache/incubator-mxnet -b v1.x mxnet`

**Comment by jbensabat** [comment]
thanks a lot
jac
